src/test/java/io/bf2/kafka/authorizer/CustomAclAuthorizerTest.java

```diff
@@ -245,6 +250,29 @@ void testCreateAclsDeniedForInvalidBinding() throws IOException {
         createAclsAndExpect("owner1", bindings, (aclCreateResult) -> isFailedWithError(aclCreateResult, CustomAclAuthorizer.CREATE_ACL_INVALID_BINDING), 3);
     }
 
+    @Test
+    void testOrphanedPrincipalsDeletedFromDelegateOnStart() throws Exception {
+        PartitionCounter partitionCounter = generateMockPartitionCounter(1001, false, false);
+        try (CustomAclAuthorizer auth = new CustomAclAuthorizer(this.delegate, partitionCounter)) {
+            auth.configure(config);
+            List<AclBindingFilter> expected = new ArrayList<>();
+            expected.add(principalFilter("User:alice"));
```

Comment:
Is `List.of(principalFilter("User:alice"), principalFilter("User:any"), principalFilter("User:bob"), principalFilter("User:anonymous"))` easier to read?

Reply:
I reworked the unit test to be more concise and readable